src/output/outputMod.F90

```diff
@@ -1823,17 +1858,93 @@ subroutine storeSurfsolInBuffer(sps, buffer, nn, blockID,   &
+             ! pp1Dim  = pp1 * pRef
+             ! pp2Dim  = pp2 * pRef
+             ! pInfDim = pInf * pRef             
+             ! pRef is defined in  module flowVarRefState (see the 'save')
+             !       but pRef is first set in the subr referenceState. However,
+             !       since it is defined in module flowVarRefState and we load
+             !       that module above it should be available to us.
+             !
+             ! Denominator: (0.5*rho*(U_a)^2) -> 
+             !       factDIM = two/(rhoDim * ((V_infDim)**2) + ((rDim*wDim)**2))
+             ! rhoDim  = rhoInfDim ! l. 81, initializeFlow.F90
+             ! V_infDim   = uInf*uRef ! also from initializeFlow.F90, l. 83
+          enddo
+          ! Comments on the computations above - no code 
+          ! pp1 and pp2 are just pressure-pointers, e.g.
+          ! pp1    => p(1,1:,1:);
+          ! Thus, they point to p and should be rescaled back to SI
+          ! as explained in the opening paragraph of this subroutine, we
+          ! have to get the average of the p in the first cell and the p in the
+          ! ghostcell/halo. This explains: half*(pp1(i,j) + pp2(i,j))
+          ! From start of subr:
+          !       As the solution
+          !       must be stored in the center of the boundary face the average
+          !       value of the first internal cell and its corresponding halo is
+          !       computed. The counter nn is updated in this routine. However
+          !       it is not initialized, because multiple contributions may be
+          !       stored in buffer.
        enddo
+       ! Comments on the computations above - no code 
+       ! [pRef] l. 19 flowVarRefState.F90
+       ! Reference pressure (in Pa) used to nondimensionalize
+       ! the flow equations. See e.g. two lines below.
+       ! [pInf] l. 81, initializeFlow.90
+       ! non-dimensional pressure
+       !    pInf   = pInfDim/pRef
+       ! [pInfDim] l. 52 flowVarRefState.F90
+       ! Free stream pressure in Pa.,
+       ! [p] l. 515 block.F90, p(:,:,:) is a pointer for static pressure
+       ! [x] l. 432 block.F90, the actual mesh for a given block. We need
+       ! these to compute the apparent velocity. Size(x) is:
+       ! x(0:ie,0:je,0:ke,3)
+       ! Well, wrt mesh it seems:
+       ! block.F90, l. 432 has the 'x'
+       ! AND
+       ! blockPointers.F90 l. 94 both have an 'x' pointer. This blockPointers
+       ! has holds the pointers for all variables inside a block.
+       !
+       ! Necessities                       units
+       ! U_i - inflow speed                      [m/s]
+       ! w   - rotational rate                   [rad/sec]
+       ! L_  - the span at point (i,j,k)         [m]
+       ! P_i - farfield pressure                 [N/m^2]
+       ! P_0 - pressure at surface point (i,j,k) [N/m^2]
+       !
+       ! fomula for apparent velocity:
+       ! (U_a)^2 = ([U_i,0,0]^T - [0,0,(2*pi*L_)/(2*pi)[m/rad]*w[rad/sec]])^2
```

Comment:
How are the units of `(2*pi*L_)/(2*pi)` `[m/rad]`? It seems like it should just be `[m]`. Why even bother with this derivation? The only part relevant for the code is the last line `(U_a)^2 = (U_i)^2 + (L_*w)^2`, and even that doesn't explain the cross-product between `L_` and `w`.